google/gax/__init__.py

```diff
@@ -482,3 +496,180 @@ def __next__(self):
         if self._index >= len(self._current):
             self._current = None
         return resource
+
+
+def _from_any(pb_type, any_pb):
+    """Converts an Any protobuf to the specified message type
+
+    Args:
+        pb_type (type): the type of the message that any_pb stores an instance
+            of.
+        any_pb (google.protobuf.any_pb2.Any): the object to be converted.
+
+    Returns:
+        An instance of the pb_type message.
+    """
+    msg = pb_type()
+    if any_pb.Unpack(msg):
+        return msg
+
+    raise TypeError(
+        'Could not convert {} to {}'.format(
+            any_pb.__class__.__name__, pb_type.__name__))
+
+
+def _try_callback(target, clbk):
+    try:
+        clbk(target)
+    except Exception as ex:  # pylint: disable=broad-except
+        _LOG.exception(ex)
+
+
+class _DeadlineExceededError(RpcError, GaxError):
+
+    def __init__(self):
+        super(_DeadlineExceededError, self).__init__('Deadline Exceeded')
+
+    def code(self):  # pylint: disable=no-self-use
+        """Always returns StatusCode.DEADLINE_EXCEEDED"""
+        return StatusCode.DEADLINE_EXCEEDED
+
+
+class _OperationFuture(object):
+    """A Future which polls a service for completion via OperationsClient."""
+
+    def __init__(self, operation, client, result_type, metadata_type,
+                 call_options=None):
+        """Constructor.
+
+        Args:
+            operation (google.longrunning.Operation): the initial long-running
+                operation object.
+            client (google.gapic.longrunning.operations_client.OperationsClient):
+                a client for the long-running operation service.
+            result_type (type): the class type of the result.
+            metadata_type (type, optional): the class type of the metadata.
+            call_options (google.gax.CallOptions, optional): the call options
+                that are used when reloading the operation.
+        """
+        self._operation = operation
+        self._client = client
+        self._result_type = result_type
+        self._metadata_type = metadata_type
+        self._call_options = call_options
+        self._done_clbks = []
+        self._process = None
+
+    def cancel(self):
+        """If last Operation's value of `done` is true, returns false;
```

Comment:
Nit: for consistency with elsewhere in GAX, don't indent the docstrings indent relative to the `"""`.